### PR TITLE
Ensure the tests directory is linted via GHA.

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
**Description**

The blanket pylint ignore all has been removed from the tests code.

In addition, the code was linted and the GitHub Actions workflow was update to include the proper linting of all relevant directories -- uwtools and tests.

Fixes Issue #85.

**Type of change**

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Pylint was run locally. GHA workflows should take care of the rest.

**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation.  I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Review by ryanlong1004 — COMMENTED** [review]
I thought we were getting rid of the blanket ```disable=all```?

**Comment by ryanlong1004** [comment]
In the __init__.py for example, you're going to get a lint error with 'unused-variables', but they aren't going to be used.  They're for namespacing.

Not sure if there's a better way to solve, but that's an issue if we remove the rule from that file.

**Comment by christinaholtNOAA** [comment]
I think we will need to leave a couple of disabled pylint rules here and there. I think it is good to make sure that the explicitly disabled ones cover the correct scope -- so, the whole file for 'unused-variables' in `__init__.py` files, for example.

**Review by TerrenceMcGuinness-NOAA — APPROVED** [review]

File: src/uwtools/files/interface/file_manager.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
-# pylint: disable=too-few-public-methods, unused-variable
+# pylint: disable=too-few-public-methods,missing-module-docstring
 
 import os
 from abc import ABC, abstractmethod
 import pathlib
 from typing import Any, List
 
 from uwtools.files.gateway import s3, unix
-from uwtools.files.model import File, Prefixes, Unix, S3
```
[the remark this answers is not in the document]
Comment (christinaholtNOAA) [reply src/uwtools/files/interface/file_manager.py:9]: That is correct.

**Comment by christinaholtNOAA** [comment]
@ryanlong1004 Did you have any other comments/questions/suggestions for this one?

**Pull request merged**